naming/src/main/java/com/alibaba/nacos/naming/controllers/InstanceController.java

```diff
@@ -139,6 +133,57 @@ public String update(HttpServletRequest request) throws Exception {
         return "ok";
     }
 
+    @CanDistro
+    @PatchMapping
+    public String patch(HttpServletRequest request) throws Exception {
+        String serviceName = WebUtils.required(request, CommonParams.SERVICE_NAME);
+        String namespaceId = WebUtils.optional(request, CommonParams.NAMESPACE_ID, Constants.DEFAULT_NAMESPACE_ID);
+        String ip = WebUtils.required(request, "ip");
+        String port = WebUtils.required(request, "port");
+        String cluster = WebUtils.optional(request, CommonParams.CLUSTER_NAME, StringUtils.EMPTY);
+        if (StringUtils.isBlank(cluster)) {
+            cluster = WebUtils.optional(request, "cluster", UtilsAndCommons.DEFAULT_CLUSTER_NAME);
+        }
+        Instance instance = serviceManager.getInstance(namespaceId, serviceName, cluster, ip, Integer.parseInt(port));
+
+        String metadata = WebUtils.optional(request, "metadata", StringUtils.EMPTY);
+        if (StringUtils.isNotBlank(metadata)) {
+            instance.setMetadata(UtilsAndCommons.parseMetadata(metadata));
+        }
+        String app = WebUtils.optional(request, "app", "DEFAULT");
+        instance.setApp(app);
+        instance.setLastBeat(System.currentTimeMillis());
+        instance.validate();
+        String weight = WebUtils.optional(request, "weight", StringUtils.EMPTY);
+        if (StringUtils.isNotBlank(weight)) {
+            instance.setWeight(Double.parseDouble(weight));
+        }
+        String healthy = WebUtils.optional(request, "healthy", StringUtils.EMPTY);
+        if (StringUtils.isNotBlank(healthy)) {
+            instance.setHealthy(BooleanUtils.toBoolean(healthy));
+        }
+        String enabledString = WebUtils.optional(request, "enabled", StringUtils.EMPTY);
+        if (StringUtils.isNotBlank(enabledString)) {
+            instance.setEnabled(BooleanUtils.toBoolean(enabledString));
+        }
+        String ephemeral = WebUtils.optional(request, "ephemeral", StringUtils.EMPTY);
+        if (StringUtils.isNotBlank(ephemeral)) {
+            instance.setEphemeral(BooleanUtils.toBoolean(ephemeral));
+        }
+
+        String agent = WebUtils.getUserAgent(request);
+
+        ClientInfo clientInfo = new ClientInfo(agent);
+
+        if (clientInfo.type == ClientInfo.ClientType.JAVA &&
```

Comment:
This logic is unnecessary, because this method should only used for 'update', not for 'registration'.